dap-mode.el

```diff
@@ -1668,10 +1674,27 @@ after selecting configuration template."
                             (funcall debug-args))
                           (user-error "Have you loaded the `%s' specific dap package?"
                                       (or (plist-get debug-args :type)
-                                          (user-error "%s does not specify :type" debug-args))))))
-    (if (functionp launch-args)
-        (funcall launch-args #'dap-start-debugging-noexpand)
-      (dap-start-debugging-noexpand launch-args))))
+                                          (user-error "%s does not specify :type" debug-args)))))
+         (cb (lambda ()
+               (if (functionp launch-args)
+                   (funcall launch-args #'dap-start-debugging-noexpand)
+                 (dap-start-debugging-noexpand launch-args)))))
+    (-if-let ((&plist :compilation) launch-args)
+        (progn
+          (cl-remf launch-args :compilation)
+          (with-current-buffer (compilation-start compilation)
+            (let (window)
+              (add-hook 'compilation-finish-functions
+                        (lambda (buf status &rest _)
+                          (if (string= "finished\n" status)
+                              (progn
+                                (when (and (not dap-debug-compilation-keep)
+                                           (window-live-p window))
+                                  (delete-window window))
```

Comment:
can we check if `get-buffer-window` in compilation buffer is non-nil and kill it in that way? Otherwise, if user displays a different buffer in that window we will still close it.

Reply:
Fixed.